src/App.js

```diff
@@ -1,28 +1,55 @@
 import './App.css'
-import Oggetti from './Components/OggettiComponents/Oggetti'
-import AllCategories from './Components/Filter/AllCategories'
-import Cart from './Components/Cart/Cart'
+import React, { useState, useEffect } from 'react'
+
 import Header from './Components/HeaderContent'
+import Shopping from './pages/Shopping'
+import Login from './pages/Login'
+import Register from './pages/Register'
+import Logout from './pages/Logout'
+
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom'
+import { checkShowMe } from './utils/api'
 
 function App() {
-  return (
-    <div>
-      <header>
-        <nav className='header_content'>
-          <Header />
-        </nav>
-      </header>
+  const [user, setUser] = useState(undefined);
 
-      <div className='aside'>
-        <AllCategories />
-        <Cart />
-      </div>
+  useEffect(() => {
+    async function getLoggedInUser() {
+      const user = await checkShowMe()
+      if (user) {
+        setUser(user)
+      }
+    }
+    getLoggedInUser()
+  }, [])
 
-      <section>
-        <div className='oggetti_container'>
-          <Oggetti />
-        </div>
-      </section>
+  function resetUser(wasLogoutSuccessful) {
+    if (wasLogoutSuccessful) {
+      setUser(undefined)
+    }
+  }
+
+  return (
+    <div>
+      <Router>
+        <header>
```

Comment:
I wrapped the header section inside the router even though it's not changing between paths, just so that we can have access to the `react-router-dom` values that we need like the `<Link>` component